lib/cc/analyzer/engine.rb

```diff
@@ -68,6 +68,11 @@ def config_file
       def output_filter
         @output_filter ||= EngineOutputFilter.new(@config)
       end
+
+      # Memory limit for a running engine in bytes
+      def memory_limit
+        (ENV["ENGINE_MEMORY_LIMIT"] || 512_000_000).to_s
```

Comment:
Do you want to suffix it with `_BYTES` so the unit is understood?
Also, maybe a constant for the default?